lib/jormungandr/src/Cardano/Wallet/Binary/Jormungandr.hs

```diff
@@ -8,5 +11,178 @@
 -- The format is for the Shelley era as implemented by the Jörmungandr node.
 
 module Cardano.Wallet.Binary.Jormungandr
-    (
+    ( getBlockHeader
+    , Message (..)
+    , BlockHeader (..)
+
+     -- * Re-export
+    , runGet
+
+    -- * For dev
+    , genesisBlock
+    , dev
     ) where
+
+import Prelude
+
+import Cardano.Wallet.Primitive.Types
+    ( SlotId (..) )
+import Data.Binary.Get
+    ( Get
+    , getByteString
+    , getWord16be
+    , getWord32be
+    , getWord8
+    , isEmpty
+    , isolate
+    , runGet
+    , skip
+    )
+import Data.ByteArray.Encoding
+    ( Base (Base16), convertFromBase )
+import Data.ByteString
+    ( ByteString )
+import Data.Word
+    ( Word16, Word32 )
+
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+
+
+-- | Messages is what the block body consists of.
+--
+-- Every message is prefixed with a message header.
+--
+--  Following, as closely as possible:
+-- https://github.com/input-output-hk/rust-cardano/blob/e0616f13bebd6b908320bddb1c1502dea0d3305a/chain-impl-mockchain/src/message/mod.rs#L22-L29
+data Message
+    = Initial [ConfigParam]
+    | OldUtxoDeclaration TODO
+    | Transaction TODO
+    | Certificate TODO
+    | UpdateProposal SignedUpdateProposal
+    | UpdateVote SignedVote
+    | UnimplementedMessage Int -- For development. Remove later.
+    deriving Show
+
+
+data BlockHeader = BlockHeader
+    { version :: Word16
+    , contentSize :: Word32
+    , slot :: SlotId
+    , chainLength :: Word32
+    , contentHash :: ByteString
+    , parentHeaderHash :: Maybe ByteString
+    } deriving (Show, Eq)
+
+-- |
+--
+-- Header
+--
+data Block = Block BlockHeader [Message]
```

Comment:
This is a different `Block` from `Cardano.Wallet.Primitive.Types.Bock`. Not sure how this will play out, but thinking we keep it here and can convert to `Cardano.Wallet.Primitive.Types.Bock` in another place as required.

Reply:
Hmmm..... Okay. we will do some mapping then most probably. Although it'd be nice if this could just happen in the decoder itself.

Reply:
In that case we will ignore everything but `Transaction` messages for now?

Reply:
Indeed. As we'll most probably make our definition of `Block` evolve when and if we take the others into consideration.